packages/inter-protocol/src/reserve/assetReserve.js

```diff
@@ -57,22 +57,13 @@ export const start = async (zcf, privateArgs, baggage) => {
     privateArgs.marshaller,
   );
 
-  /** @type {() => Promise<ZCFMint<'nat'>>} */
-  const takeFeeMint = async () => {
-    if (baggage.has('feeMint')) {
-      return baggage.get('feeMint');
-    }
-
-    const feeMintTemp = await zcf.registerFeeMint(
-      'Fee',
-      privateArgs.feeMintAccess,
-    );
-    baggage.init('feeMint', feeMintTemp);
-    return feeMintTemp;
-  };
-  trace('awaiting takeFeeMint');
-  const feeMint = await takeFeeMint();
   const storageNode = await privateArgs.storageNode;
+
+  trace('awaiting feeMint');
+  const { feeMint } = await provideAll(baggage, {
+    feeMint: () => zcf.registerFeeMint('Fee', privateArgs.feeMintAccess),
+  });
+
```

Comment:
Assuming the `feeMint` wasn't already in the `baggage`, which is why `takeFeeMint()` was failing, how did `provideAll` solved the "_no contact with other vats during the first crank of the new incarnation_"? Does `provideAll` somehow delay the execution of `zcf.registerFeeMint` to another crank? 

cc @Chris-Hibbert

Reply:
`feeMint` was already in baggage. It was failing because we were upgrading the governor contract instead of the reserve contract, which I assume has a different baggage attached to it. `provideAll` just abstracts the logic of creating the item in baggage on the first incarnation and then checking the value on future upgrades.